bodo/libs/_memory.cpp

```diff
@@ -703,10 +703,27 @@ BufferPoolOptions BufferPoolOptions::Defaults() {
         options._allocate_extra_frames = true;
     } else {
         // Fraction of total memory we should actually assign to the buffer
-        // pool. We will read this from an environment variable if it's set, but
-        // will default to 95% if spilling is available and enabled, and to 500%
-        // otherwise.
-        double mem_fraction = options.storage_options.empty() ? 5. : .95;
+        // pool.
+        // - 0.8 is the default value.
+        // - If we're running on a remote system, we use 0.95 with spilling
+        //   enabled, 5.0 without spilling enabled.
+        // - If users set an environment variable, we use that fraction.
+        double mem_fraction = 0.8;
```

Comment:
Isn't this way too low? We usually encountered significant underutilization of the system's physical memory. That's why we set 5 as the default I believe.

One factor is streaming since buffers double in size repeatedly but a lot of the memory isn't actually touched. We will use streaming in Python later too though.

Reply:
So my concern is that on local systems like a laptop, people are not going to run Bodo by itself. They usually will have a web browser open with a couple of tabs, maybe VSCode or JupyterLab, and some other stuff. That already incurs a lot of memory utilization and swap space. For example, on my laptop having VSCode open and 4 Firefox tabs (2 of which are just static pages) had about 4GB / 16GB in memory and 4GB swap. Now I do have more background things than others, like I used a fancier LSP in VSCode in all that. But I don't think 80% is unreasonable.

Reply:
When we start integrating streaming with Python, we can check back on this, but we cant just assume that Bodo will be the only CPU and memory intensive process. We may need to invest time working on reducing the amount of underutilized memory, like via segmented data structures (https://github.com/martinus/unordered_dense?tab=readme-ov-file#4-segmented_map-and-segmented_set) or array packing in buffers.

Reply:
Oh plus by that point we need to figure out the spilling config anyways, so it might be a moot point

Reply:
80% of physical memory is not unreasonable. My concern is that our logical size may end up 20% of physical memory for example.

Reply:
What do others do?

Reply:
Maybe I should clarify, I'm opening up a PR on the platform to set the env `BODO_BUFFER_POOL_REMOTE_MODE=1` on the AMI, so the platform is unaffected. Plus, on benchmarking, we can use that env.

Reply:
DuckDB does the exact same thing: https://duckdb.org/docs/operations_manual/limits.html. 80% of RAM. But I know that they've put in the effort to improve the memory packing.